### ISSUE TITLE
Switch kubectl contexts via component

### ISSUE THREAD

**Issue by daneah** [daneah/iterm-components#1]
It would be awesome if the kubectl context component, when clicked, would show a popover menu of the available contexts via:

```
kubectl config view -o jsonpath='{.contexts[*].name}' | tr " " "\n"
```

And switch to one of the displayed contexts when clicked via:

```
kubectl config use-context <context>
```

The popover is covered well in the docs (see `onclick` in [this example](https://iterm2.com/python-api/examples/jsonpretty.html#jsonpretty-example)) but being able to click items inside the popover isn't AFAICT.

The built in `git state` component does something like this, but the built-ins all appear to be written in Objective-C.

**Comment by gragera** [comment]
I don't think this can be done just yet. The python API does not offer any way of creating menu items, selecting them, etc.

The python API will most likely evolve and add the ability to handle menus, but considering the  current state, doing the context selection does not seem feasible :)

Nice work by the way!

**Comment by daneah** [comment]
Thanks for the input @gragera. I wondered if this might be possible using the popover and some further response to a click action inside the popover, but I didn't get far enough on it to say one way or the other.
